test/advi_locscale.jl

```diff
@@ -0,0 +1,86 @@
+
+const PROGRESS = length(ARGS) > 0 && ARGS[1] == "--progress" ? true : false
+
+using Test
+
+@testset "advi" begin
+    @testset "locscale" begin
+        @testset "$(modelname) $(objname) $(realtype) $(adbackname)"  for
+            realtype ∈ [Float64], # Currently only tested against Float64
+            (modelname, modelconstr) ∈ Dict(
+                :NormalLogNormalMeanField => normallognormal_meanfield,
+            ),
+            (objname, objective) ∈ Dict(
+                :ADVIClosedFormEntropy  => ADVI(10),
+                :ADVIStickingTheLanding => ADVI(10, entropy = StickingTheLandingEntropy()),
+            ),
+            (adbackname, adbackend) ∈ Dict(
+                :ForwarDiff  => AutoForwardDiff(),
+                # :ReverseDiff => AutoReverseDiff(),
+                # :Zygote      => AutoZygote(), 
+                # :Enzyme      => AutoEnzyme(),
+            )
+
+            seed = (0x38bef07cf9cc549d)
```

Comment:
I'd recommend against using such an obscure number :sweat_smile: Is there a reason for it?

Reply:
This is because it's good practice in general to use "full" seeds with random-looking bits instead of small integers like `123`. I could use a more common simple seed though, since these are just tests.

Reply:
Oh wow, this is not something I was aware of! Thanks for pointing that out:)

But a couple of things:
- Julia actually uses [Xoshiro256++ by default](https://docs.julialang.org/en/v1/stdlib/Random/), which, if I've understood correctly, should be more resilient to such things.
- `@testset` automatically sets the random seed, explicit `seed!` shouldn't really be necessary unless there a very unstable tests where we encounter failures only on some devices (in this case `StableRNG` might be useful).

Reply:
Oh wow, this is not something I was aware of! Thanks for pointing that out:)

But a couple of things:
- Julia actually uses [Xoshiro256++ by default](https://docs.julialang.org/en/v1/stdlib/Random/), which, if I've understood correctly, should be more resilient to such things.
- `@testset` automatically sets the random seed, explicit `seed!` shouldn't really be necessary unless there a very unstable tests where we encounter failures only on some devices (in this case `StableRNG` might be useful).

Reply:
Ah, I see you're testing that it's deterministic below, so then it makes sense:)

Reply:
> @testset automatically sets the random seed, explicit seed!
Oh, I see! That's actually very nice. I didn't imagine it would do that. Then, I might clean up the tests that don't actually need explicit seeding, but I guess it's not urgent for now.